src/pages/Dashboard/MyAccount/MyAccountUi.tsx

```diff
@@ -0,0 +1,217 @@
+/** @jsxImportSource @emotion/react */
+import React from 'react';
+import Typography from '@mui/material/Typography';
+
+import { formatCentsToReadableValue } from 'utilities/common';
+import { useTranslation } from 'translation';
+import { IToggleProps, Toggle, Icon, ProgressBarHorizontal, Tooltip } from 'components';
+import { useMyAccountStyles as useStyles } from './styles';
+
+export interface IMyAccountUiProps {
+  netApyPercentage: number | undefined;
+  dailyEarningsCents: number | undefined;
+  supplyBalanceCents: number | undefined;
+  borrowBalanceCents: number | undefined;
+  borrowLimitCents: number | undefined;
+  safeBorrowLimitPercentage: number;
+  withXvs: boolean;
+  onXvsToggle: (newValue: boolean) => void;
+  className?: string;
+}
+
+export const MyAccountUi = ({
+  netApyPercentage,
+  dailyEarningsCents,
+  supplyBalanceCents,
+  borrowBalanceCents,
+  borrowLimitCents,
+  safeBorrowLimitPercentage,
+  withXvs,
+  onXvsToggle,
+  className,
+}: IMyAccountUiProps) => {
+  const styles = useStyles();
+  const { t, Trans } = useTranslation();
+
+  const handleXvsToggleChange: IToggleProps['onChange'] = (_event, checked) => onXvsToggle(checked);
+
+  const readableBorrowBalance =
+    typeof borrowBalanceCents === 'number'
+      ? formatCentsToReadableValue(borrowBalanceCents)
```

Comment:
Since these functions produce a readable value, I'd include the undefined check inside the function and return the `'-'` as the readable value if `undefined` or unparseable

Reply:
Perhaps the naming is wrong, but we need it to be undefined if `borrowBalanceCents` is not a number so that we can reused it for the `trackTooltip` prop of `ProgressBarHorizontal`.

I could use the prefix `formatted` rather than `readable`?

Reply:
You could also use the a constant for the empty value and refuse the tooltip if it is the empty content. Alternatively, we could skip the empty formatted value here. I'll comment other places where the default formatted logic is also duplicated.

Reply:
A better option would be to wait to format until you've decided to render the tooltip or not use the original value to decided to render the tooltip `borrowBalanceCents`

Reply:
I see what you mean, I like that idea.
I'll add the constant, but also create a util that can be reused throughout the app to render values that can be `undefined`.

About the tooltip, we use this formatted value both in the component itself and the tooltip. We only render the tooltip if the `borrowBalanceCents` isn't `undefined`, hence why `readableBorrowBalance` needs to best casted as `undefined` if that is the case. I know its confusing, I think the util will help streamline all that.

Reply:
I made some updates that should clear things up.
I've updated the `formatCentsToReadableValue` so that it accepts an `undefined` value and returns a placeholder (defined as a constant; `PLACEHOLDER_KEY`) in that case. I figured this will can be used a lot throughout the codebase. I also updated `formatToReadablePercentage` to follow the same behavior.

This meant I could strip out a lot of duplicated logic from my PR and use these utils instead.
Let me know if that's clearer now.